### PR TITLE
Annual electricity use per pupil with savings potential

### PR DESCRIPTION
**Pull request by urbanwide**

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -60,6 +60,7 @@
   end
 
   namespace :comparisons do
+    resources :annual_electricity_costs_per_pupil, only: [:index]
```
Comment (tbhi) [inline config/routes.rb:63]: yeah we should probably sort these, I've just been adding to the bottom :grimacing:

**Review by tbhi — APPROVED** [review]
looks good to me, don't know if @ldodds also wants to review?

**Review by ldodds — APPROVED** [review]

File: config/routes.rb (hunk 1 above)
Comment (urbanwide) [reply config/routes.rb:63]: The route generator seems to put these at the top of the namespace by default.. (now it is working - see https://github.com/Energy-Sparks/energy-sparks/pull/3472)

File: app/models/comparison/annual_electricity_costs_per_pupil.rb [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+#
+#  alert_generation_run_id                    :bigint(8)
+#  id                                         :bigint(8)
+#  last_year_gbp                              :float
+#  one_year_electricity_per_pupil_gbp         :float
+#  one_year_saving_versus_exemplar_gbpcurrent :float
+#  school_id                                  :bigint(8)
+#
+class Comparison::AnnualElectricityCostsPerPupil < Comparison::View
+  scope :with_data, -> { where.not(one_year_electricity_per_pupil_gbp: nil) }
```
Comment (urbanwide) [inline app/models/comparison/annual_electricity_costs_per_pupil.rb:13]: The original doesn't have this condition but it seems like a good idea to make sure there is some data before displaying the school?

Comment (ldodds) [reply app/models/comparison/annual_electricity_costs_per_pupil.rb:13]: Yes, seems sensible

**Review by ldodds — APPROVED** [review]

**Pull request merged**